### PR TITLE
Use PHP's built-in sort algorithms

### PR DESCRIPTION
**Pull request by darthsteven**
https://github.com/vanderlee/PHP-stable-sort-functions/blob/master/classes/StableSort.php#L128

### CODE REVIEW

**Comment by darthsteven** [comment]
@stevetweeddale any chance of a review?

File: cm_tools.module [hunk 1]
```diff
@@ -328,41 +328,24 @@ function cm_tools_array_remove_values(&$haystack, $values) {
...
+  foreach ($array as &$item) {
+    $item = array($index++, $item);
+  }
+  $result = usort($array, function($a, $b) use($cmp_function) {
+    $result = call_user_func($cmp_function, $a[1], $b[1]);
+    return $result == 0 ? $a[0] - $b[0] : $result;
+  });
+  foreach ($array as &$item) {
+    $item = $item[1];
+  }
```
Comment (stevetweeddale) [inline cm_tools.module:341]: This is such a simple and clever way of doing this. Three cheers for whoever that guy was.

**Review by stevetweeddale — APPROVED** [review]
This is approximately a bazillion times betterer.

**Comment by darthsteven** [comment]
I've made the requirement for PHP 5.3 explicit in the info file.

**Pull request merged**